src/hotspot/share/opto/stringopts.cpp

```diff
@@ -1022,6 +1022,20 @@ bool StringConcat::validate_control_flow() {
       fail = true;
       break;
     } else if (ptr->is_Proj() && ptr->in(0)->is_Initialize()) {
+      // Check for side effect between Initialize and the constructor
+      for (SimpleDUIterator iter(ptr); iter.has_next(); iter.next()) {
+        Node* use = iter.get();
+        if (!use->is_CFG() && !use->is_CheckCastPP() && !use->is_Load()) {
+#ifndef PRODUCT
+          if (PrintOptimizeStringConcat) {
+            tty->print_cr("unexpected control use of Initialize");
+            use->dump(2);
```

Comment:
What output of `dump(2)` you got in your case? It could be more than needed if `use` has a lot of inputs.
How about next to output only interesting info?:
```
ptr->in(0)->dump(); // Initialize node
use->dump(1);
```